src/System.Collections/src/System/Collections/BitArray.cs

```diff
@@ -341,6 +341,123 @@ public BitArray Not()
             return this;
         }
 
+        /*=========================================================================
+        ** Shift all the bit values to right on count bits. When count less zero it
+        ** shift all the bit values to left. New bits are be set to zero. The 
+        ** current instance is updated and returned.
+        =========================================================================*/
+        public BitArray RightShift(int count)
+        {
+            if (count <= 0)
+            {
+                if (count == 0)
+                {
+                    _version++;
+                    return this;
+                }
+                count = -count;
+                // Following trick reduce check with int.MinValue
+                if (unchecked((uint)count < (uint)m_length))
+                    return LeftShift(count);
+            }
+
+            int toIndex = 0;
+            int ints = GetArrayLength(m_length, BitsPerInt32);
+            if (unchecked((uint)count < (uint)m_length))
+            {
+                var fromIndex = count / BitsPerInt32;
+                int shiftCount = count % BitsPerInt32;
```

Comment:
Perhaps you could use Math.DivRem here? Good discussion about jit behaviour here https://github.com/dotnet/coreclr/issues/3439